modules/websocket_bridge/include/hephaestus/ipc/ipc_interface.h

```diff
@@ -0,0 +1,102 @@
+#include <span>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <absl/base/thread_annotations.h>
+#include <absl/synchronization/mutex.h>
+#include <hephaestus/ipc/topic.h>
+#include <hephaestus/ipc/zenoh/raw_publisher.h>
+#include <hephaestus/ipc/zenoh/raw_subscriber.h>
+#include <hephaestus/ipc/zenoh/service.h>
+#include <hephaestus/ipc/zenoh/session.h>
+#include <hephaestus/serdes/type_info.h>
+
+namespace heph::ws {
+
+using TopicSubscriberWithTypeCallback = std::function<void(
+    const ipc::zenoh::MessageMetadata&, std::span<const std::byte>, const serdes::TypeInfo&)>;
+
+using RawServiceResponses = std::vector<ipc::zenoh::ServiceResponse<std::vector<std::byte>>>;
+using AsyncServiceResponseCallback = std::function<void(const RawServiceResponses&)>;
+
+class IpcInterface {
+public:
+  IpcInterface(std::shared_ptr<ipc::zenoh::Session> session, ipc::zenoh::Config config);
+
+  ~IpcInterface();
+
+  void start();
+  void stop();
+
+  // Subsribers
+  /////////////
+  [[nodiscard]] auto hasSubscriber(const std::string& topic) const -> bool;
+  void addSubscriber(const std::string& topic, const serdes::TypeInfo& topic_type_info,
+                     const TopicSubscriberWithTypeCallback& subscriber_cb);
+  void removeSubscriber(const std::string& topic);
+
+  // Publishers
+  /////////////
+  [[nodiscard]] auto hasPublisher(const std::string& topic) const -> bool;
+  void addPublisher(const std::string& topic, const serdes::TypeInfo& topic_type_info);
+  void removePublisher(const std::string& topic);
+  [[nodiscard]] auto publishMessage(const std::string& topic, std::span<const std::byte> data) -> bool;
+
+  // Services
+  ///////////
+  [[nodiscard]] auto callService(uint32_t call_id, const ipc::TopicConfig& topic_config,
+                                 std::span<const std::byte> buffer, std::chrono::milliseconds timeout)
+      -> RawServiceResponses;
+  auto callServiceAsync(uint32_t call_id, const ipc::TopicConfig& topic_config,
+                        std::span<const std::byte> buffer, std::chrono::milliseconds timeout,
+                        AsyncServiceResponseCallback callback) -> std::future<void>;
+
+private:
+  std::shared_ptr<ipc::zenoh::Session> session_;
+  ipc::zenoh::Config config_;
+
+  // Subscribers
+  //////////////
+  mutable absl::Mutex mutex_sub_;
+  std::unordered_map<std::string, std::unique_ptr<ipc::zenoh::RawSubscriber>>
+      subscribers_ ABSL_GUARDED_BY(mutex_sub_);
+
+  // NOLINTNEXTLINE(modernize-use-trailing-return-type)
+  [[nodiscard]] bool hasSubscriberImpl(const std::string& topic) const
+      ABSL_EXCLUSIVE_LOCKS_REQUIRED(mutex_sub_);
+
+  // Publishers
+  /////////////
+  mutable absl::Mutex mutex_pub_;
+  std::unordered_map<std::string, std::unique_ptr<ipc::zenoh::RawPublisher>>
+      publishers_ ABSL_GUARDED_BY(mutex_pub_);
+
+  // NOLINTNEXTLINE(readability-identifier-naming)
+  static void callback_PublisherMatchingStatus(const std::string& topic,
```

Comment:
```suggestion
  static void publisherMatchingStatusCallback(const std::string& topic,
```

Reply:
I quite like the increase in readability this clear  callback_* naming brings. I liked it enough to tell the linter to ignore it. But I guess I'll rename it ;)

Reply:
But can we please keep the callback naming in bridge.h / bridge.cpp
It just makes it so much more intutitive not only to recognize callbacks immediately, but see who will be calling them from a architecture PoV.